### ISSUE TITLE
Enhancement: Handle pagination in JSON 

### ISSUE THREAD

**Issue by arky** [CJWorkbench/loadurl#2]
There are many APIs that have inbuilt pagination with "next" and "previous" fields. It would great to have functionality to handle similar to pagination in scraping modules.  

Sample API:  http://api.openhluttaw.org/en/persons

**Comment by arky** [comment]
Response from @adamhooper via slack. Adding this here for documentation purposes. 

1. User-interface complexity. The "obvious" way of paginating -- "page size" + "page number" (like the page you linked to) adds duplicate items or skips items when somebody modifies the list while somebody else is paginating through the API. (And what's the point in an API if the data it accesses never changes!) Correct APIs (like Facebook/Twitter) use a "cursor" strategy. All these different strategies are complex. (Then there's the "retry" problem: among 200 requests, one or two are _bound_ to fail and should be retried....) Workbench needs a way to help users figure out the correct way to paginate through their API. Alternatively, the community can create one module per API.

2. Slowness. If an API takes 15s to respond, and there are 200 pages, that's 50min of fetching. Workbench doesn't permit long-running modules yet. (Timeout is currently 10min.) We need to implement A) a paywall (so we can abort fetches that take too long); and B) a way for modules to report their progress to end users via a progressbar interface (so users don't think Workbench is broken).
In sum:
* We'd love to help create a module that fetches from your API. I can point you to Workbench libraries that can help; and we'd love to expand those libraries to help module developers handle retries and concatenation.
* 38 pages @ 12s/page means ~8 minutes to fetch data from this API. Workbench's timeout is currently set to 10min. Can you increase the page size? Otherwise, the Workbench module is likely to time out some fraction of the time.

**Comment by adamhooper** [comment]
@arky Thank you for the request!

Let's narrow this report down to the one API you mention. "Solving" all Pagination APIs is akin to "solving" HTML scraping: we'll never be done.

**Comment by arky** [comment]
Fair enough @adamhooper  Perhaps I could reach out to Openhluttaw team to ask them to implement '[cursor](https://pypi.org/project/django-cursor-pagination/)' based pagination. Am sure if Django REST framework supports that out of the box.
